darwin/future/meta/objects/stage.py

```diff
@@ -59,7 +61,52 @@ def item_ids(self) -> ItemIDQuery:
             ],
         )
 
-    def move_attached_files_to_stage(self, new_stage_id: UUID) -> Stage:
+    def check_all_items_complete(
+        self,
+        slug: str,
+        item_ids: list[UUID],
+        wait_max_attempts: int = 5,
+        wait_time: float = 0.5,
+    ) -> bool:
+        """
+        Checks if all items are complete. If not, waits and tries again. Raises error if max attempts reached.
+
+        Args:
+            slug (str): Team slug
+            item_ids (list[UUID]): List of item ids
+            max_attempts (int, optional): Max number of attempts. Defaults to 5.
+            wait_time (float, optional): Wait time between attempts. Defaults to 0.5.
+        """
+        completed = []
+        for attempt in range(1, wait_max_attempts + 1):
+            # check if all items are complete
+            for item_id in item_ids:
+                if get_item(self.client, slug, item_id).processing_status != "complete":
+                    # if not complete, wait and try again with the in-complete items
+                    time.sleep(wait_time * attempt)
```

Comment:
this time.sleep should be outside of the item_ids loop, we want it to wait a certain amount of time after checking the entire loop rather than per item. Even a small wait per item is going to add up when looking at 1000+ items and each check is independent of the others in the list anyway. Also can just do item_ids.remove(id) rather than the list comprehension every time. Either works though.

Reply:
Good catch! Fixed in a new commit.

>  can just do item_ids.remove(id) rather than the list comprehension every time

This is done as well - It is now iterating over a copy of `item_ids[:]` and removing the item from `item_ids`.